greenheart/tools/eco/electrolysis.py

```diff
@@ -33,37 +33,62 @@ def run_electrolyzer_physics(
     electrolyzer_size_mw = greenheart_config["electrolyzer"]["rating"]
     electrolyzer_capex_kw = greenheart_config["electrolyzer"]["electrolyzer_capex"]
     
+    # IF GRID CONNECTED
     if greenheart_config["project_parameters"]["grid_connection"]:
-        energy_to_electrolyzer_kw = np.ones(365 * 24 - 4*7*12) * ( # TODO why the subtraction here?
-            electrolyzer_size_mw * 1e3
-        )
+        n_pem_clusters = 1
+        #Eco-grid conection cases
+        if greenheart_config["project_parameters"]["grid_to_electrolyzer_input"] == "power":
+            energy_to_electrolyzer_kw = np.ones(365 * 24 - 4*7*12) * ( # TODO why the subtraction here?
+                electrolyzer_size_mw * 1e3
+            )
+            grid_connection_scenario='off-grid'
+            hydrogen_production_capacity_required_kgphr=0
+        #GS grid-connected case
+        else:
+            # n_pem_clusters = 1
+            energy_to_electrolyzer_kw = np.zeros(8760)
+            grid_connection_scenario='grid-only'
+            hydrogen_production_capacity_required_kgphr=greenheart_config["electrolyzer"]["hydrogen_dmd"]
+
+
+
     else:
+        hydrogen_production_capacity_required_kgphr = 0 
+        grid_connection_scenario='off-grid'
         energy_to_electrolyzer_kw = np.asarray(hopp_results[
             "combined_hybrid_power_production_hopp"
         ])
+        if electrolyzer_size_mw % greenheart_config["electrolyzer"]["cluster_rating_MW"] == 0:
```

Comment:
I'm not sure I totally follow the logic here. Can you add a few comments of explanation?

Reply:
In the code for future users